design/014/API.md

```diff
@@ -0,0 +1,142 @@
+# API Design for Out-of-Process Providers
+
+## Problem Description
+
+How should `ExternalSecret` and `PushSecret` resources reference out-of-process providers?
+
+How would the equivalent of `ClusterSecretStore` look like?
+
+## Context
+
+Out-of-process providers run in separate pods and communicate with ESO Core via gRPC. We need an API design that enables:
+
+- Service discovery: ESO Core must locate the provider's gRPC endpoint
```

Comment:
I think to ease this, it should be the other way around, right? Meaning, the plugin should discover ESO instead. Since that can be determined during installation ( with a Service ). WDYT?